pulsar-client-cpp/lib/PartitionedConsumerImpl.cc

```diff
@@ -106,6 +106,9 @@ Result PartitionedConsumerImpl::receive(Message& msg, int timeout) {
         unAckedMessageTrackerPtr_->add(msg.getMessageId());
         return ResultOk;
     } else {
+        if (state_ != Ready) {
+            return ResultAlreadyClosed;
+        }
```

Comment:
Currently, we need to add a lock here because `state_` is not atomic. I think we can change it to an atomic variable in another PR.

Reply:
Thanks，I will fix it